### ISSUE TITLE
load_repository(): fails to load "0.json" metadata file

### ISSUE THREAD

**Issue by jku** [theupdateframework/tuf#1069]
If you have a repository with metadata files 0.json through 9.json they will fail to load in `load_repository()` call seemingly because `_strip_version_number()` mangles them.

These metadata files can be created e.g. by delegating to 16 hashed bins.

**Current behavior**:
```
>>> repository=load_repository("repository")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/jku/src/tuf/tuf/repository_tool.py", line 3038, in load_repository
    metadata_path = delegated_roles_filenames[rolename]
KeyError: '0'

```

**Comment by joshuagl** [comment]
I think the bug here is that `repository_lib._load_top_level_metadata()` is setting _consistent_snapshot_ to true when the repository isn't in fact using consistent snapshots. This in turn is causing `_strip_version_number()` to try and strip a version number from the beginning of a path which doesn't have one.

As well as fixing whatever issue is causing `_strip_version_number()` to be passed `consistent_snapshot=True` we should also update `_strip_version_number()` to ensure the passed filename has 3 components before stripping anything.

**Comment by lukpueh** [comment]
Fixed by #1071
